hvac/v1/__init__.py

```diff
@@ -978,6 +986,156 @@ def auth_approle(self, role_id, secret_id=None, mount_point='approle', use_token
 
         return self.auth('/v1/auth/{0}/login'.format(mount_point), json=params, use_token=use_token)
 
+    def auth_iam(self, iam_role):
+        """
+        Authenticate to Vault per https://www.vaultproject.io/docs/auth/aws.htm.
+
+        :param iam_role: IAM role
+        :return:
+        """
+        parsed = urlparse(self._url)
+        self.token = self._authenticate_to_vault(parsed.hostname,
+                                                 parsed.port,
+                                                 iam_role,
+                                                 self._kwargs['verify'])
+        return self.token
+
+    def _authenticate_to_vault(self, vault_host, vault_port, role, verify):
```

Comment:
This is duplicative of other methods already extant in the library and should be dropped. E.g., the `auth_ec2()` method: https://github.com/ianunruh/hvac/blob/master/hvac/v1/__init__.py#L593-L603. I would recommend looking at the various other auth_ methods and making sure you are following the same pattern.